src/components/JoinGroupForm/JoinGroupForm.tsx

```diff
@@ -70,10 +68,65 @@ export default function JoinGroupForm() {
 
   const handleClick = async (event: React.MouseEvent<HTMLButtonElement>) => {
     event.preventDefault();
+    dispatch(setLoading());
+
+    try {
+      const data = await getClass(user?.authToken as string, classCode);
+
+      if (data.status === 'success') {
+        setClassInfo(data.data as Class);
+      } else {
+        dispatch(
+          updateToast({
+            title: data.status,
+            type: 'error',
+            message: data.data as string,
+          })
+        );
+      }
+    } catch (error) {
+      console.log(error);
+      dispatch(updateToast(GENERAL_ERRORS.SYSTEM));
+    }
+    dispatch(removeLoading());
+  };
 
-    const data = await dummyInfo<ClassInfo>();
-
-    setClassInfo(data);
+  const joinClassHandler = async (
```

Comment:
For consistency, if above we are naming a handler 'handleSomething' I think this should be 'handleJoinClass'. 
I know we are probably having different naming conventions among files, but I think within files we should try to stick to one.